src/vs/code/electron-main/backup.ts

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as fs from 'original-fs';
+import * as path from 'path';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+
+export const IBackupService = createDecorator<IBackupService>('backupService');
+
+export interface IBackupService {
+	getBackupWorkspaces(): string[];
+	clearBackupWorkspaces(): void;
+	pushBackupWorkspaces(workspaces: string[]): void;
```

Comment:
Not quite clear here what `workspaces` is about? Is this an identifier or absolute path?